opm/parser/eclipse/Parser/Parser.hpp

```diff
@@ -130,20 +131,21 @@ namespace Opm {
         const std::vector<std::pair<std::string,std::string>> codeKeywords() const;
 
     private:
-        // associative map of the parser internal name and the corresponding ParserKeyword object
-        std::vector< std::unique_ptr< const ParserKeyword > > keyword_storage;
+        bool hasWildCardKeyword(const std::string& keyword) const;
+        const ParserKeyword* matchingKeyword(const string_view& keyword) const;
+        void addDefaultKeywords();
+
+        // std::vector< std::unique_ptr< const ParserKeyword > > keyword_storage;
+        std::list<ParserKeyword> keyword_storage;
```

Comment:
Do we ever iterate over the `keyword_storage`?  If so, did you try to measure if there is a performance impact from switching this from a `std::vector` to a `std::list`?

Reply:
Thanks for looking into this.

> Do we ever iterate over the keyword_storage?

No - lookup is solely based on different varieties of map based lookup. The maps are of type:
```
std::map<std::string, const ParserKeyword *>
```
where the pointers point into the storage managed by the linear container. Previously that pointed into the `unique_ptr::get()` - now it points directly into the container - and we *need stable storage* - that is the reason for changing `std::vector<>` &rightarrow; `std::list<>`.

Reply:
> No - lookup is solely based on different varieties of map based lookup. The maps are of type:
> 
> ```C++
> std::map<std::string, pointer>
> ```
> [...] and we need stable storage [for the items].

I see&mdash;that makes sense.  Depending on this structure is actually used, we might alternatively consider a structure akin to
```C++
using KwPtr   = std::unique_ptr<const ParserKeyword>;
using Storage = std::vector<KwPtr>;
using Ix      = Storage::size_type;

std::map<std::string, Ix> keyword_storage;
```
since vector indices aren't affected by `vector<>::push_back()`'s iterator invalidation behaviour.

Still, that's only something to keep in the back of our minds for any future reorganisation.

Reply:
> Depending on how this structure is actually used, we might alternatively consider a structure akin to [...]


I agree - your approach is better than the current; another day though.